include/intx/intx.hpp

```diff
@@ -1943,6 +1943,24 @@ inline constexpr uint<N>& operator>>=(uint<N>& x, const T& y) noexcept
 
 inline uint256 addmod(const uint256& x, const uint256& y, const uint256& mod) noexcept
 {
+    // Fast path for mod >= 2^192, with x and y at most slightly bigger than mod.
+    // This is always the case when x and y are already reduced modulo mod.
+    // Based on https://github.com/holiman/uint256/pull/86.
+    if ((mod[3] != 0) && (x[3] <= mod[3]) && (y[3] <= mod[3]))
```

Comment:
How does this handle the `mod[4] != 0` case? i.e. shouldn't it be `(mod[4] | mod[3]) != 0`?

Reply:
There is no `mod[4]`. The `mod[3]` is the most significant word.

Reply:
You are correct, phew :)